src/rust/perftools/profiler/mod.rs

```diff
@@ -302,12 +307,29 @@ impl Profiler {
         let total_duration = self.roots.iter().map(|root| root.borrow().duration_sum).sum();
 
         for root in self.roots.iter() {
-            root.borrow().write_recursive(out, total_duration, 0, max_depth)?;
+            root.borrow().write_recursive(out, total_duration, 0, max_depth, self.ns_per_cycle)?;
         }
 
         out.flush()
     }
 
+    fn measure_ns_per_cycle() -> f64 {
+        let start = SystemTime::now();
```

Comment:
Could you please annotate the following variable declaration with the concerned type?